File: programming files/rgb_to_hex.java
// 04/01/2025 Weds - Sean

public class rgb_to_hex {
    public static void main(String[] args) {
        // main file that sets r, g, and b values
        int r = 255;
        int g = 255;
        int b = 255;
        // creates a string named hexColor using the function rgbToHex
        String hexColor = rgbToHex(r, g, b);
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor);
    }

    public static String rgbToHex(int r, int g, int b) {
        // this function math.min and math.max
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.max(0, g));
        b = Math.min(255, Math.max(0, b));
        return String.format("%02X%02X%02X", r, g, b);
    }
}

//Test with RGB color (254, 127, 0) = FF7F00

File: programming files/rgb_to_hex.cc
#include <iostream>
#include <sstream>
#include <iomanip>
//Take the color values in RGB and converts to a hexadecimal format
std::string rgb_to_hex(int r, int g, int b)
{
    //sets string to rgb  values
    r = std::max(0, std::min(255, r));
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));

    //converts to hexadecimal
    std::stringstream ss;
    ss << std::uppercase << std::hex << std::setfill('0')
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;

    return ss.str();
}

//Test with std::string hexColor = rgb_to_hex(255, 127, 0); // returns "FF7F00"
